### ISSUE TITLE
Exposing the transpose kernel to clients

### ISSUE THREAD

**Issue by blueberry** [CNugteren/CLBlast#67]
Hi  Cedric,

Is there a reason, other than not being in BLAS standard, to not expose transpose functionality to the clients of CLBlast?

Since CLBlast has this, and, probably, other useful utility functions that are already implemented, would you consider making those available through a public API?

**Comment by CNugteren** [comment]
No, not really. I guess it makes sense indeed to expose the transpose kernel to the user. Already a couple of other non-BLAS functions are available. I also know that cuBLAS has additional functions (http://docs.nvidia.com/cuda/cublas/#blas-like-extension) but it doesn't seem that transpose is included there.

I'll look into this.

**Comment by blueberry** [comment]
Thanks! That could be quite helpful :)

**Comment by blueberry** [comment]
Another good candidate function is, I think, a matrix copy function that takes into account `ld` and `order`.

**Comment by CNugteren** [comment]
I believe the `omatcopy` family or routines is what you request for: https://software.intel.com/en-us/node/468656

I'll work on implementing this, which includes both out-of-place copy, transpose, and scaling.

**Comment by blueberry** [comment]
Hey, that's a good find :) Thank you for that implementation.

**Comment by CNugteren** [comment]
It's implemented:
```
- Added non-BLAS routines:
  * SOMATCOPY/DOMATCOPY/COMATCOPY/ZOMATCOPY/HOMATCOPY (matrix copy, scaling, and/or transpose)
```
This is the first non-BLAS routine for which I also included tests, so I had to re-organise some things in the library. I haven't tested it against MKL, but I implemented it according to their documentation, so hopefully it is exactly the same routine.

It is currently in the `development` branch, but will automatically be included in the next release.
